File: Backend/internal/entity/job.go
package entity

import (
	"time"

	"github.com/google/uuid"
)

type Job struct {
	ID                uuid.UUID  `db:"id" json:"id"`
	ClientID          uuid.UUID  `db:"client_id" json:"client_id"`
	Title             string     `db:"title" json:"title"`
	Summary           string     `db:"summary" json:"summary"`
	Description       string     `db:"description" json:"description"`
	Category          string     `db:"category" json:"category"`
	Skills            []string   `db:"skills" json:"skills"`
	BudgetType        string     `db:"budget_type" json:"budget_type"`
	BudgetMin         float64    `db:"budget_min" json:"budget_min"`
	BudgetMax         float64    `db:"budget_max" json:"budget_max"`
	DurationLabel     string     `db:"duration_label" json:"duration"`
	ExperienceTier    string     `db:"experience_tier" json:"experience"`
	Deliverables      []string   `db:"deliverables" json:"deliverables"`
	Urgent            bool       `db:"urgent" json:"urgent"`
	Status            string     `db:"status" json:"status"`
	ProposalsCount    int        `db:"proposals_count" json:"proposals"`
	HiredFreelancerID *uuid.UUID `db:"hired_freelancer_id" json:"hired_freelancer_id,omitempty"`
	Aspect            *string    `db:"aspect" json:"aspect,omitempty"`
	PostedAt          time.Time  `db:"posted_at" json:"posted_at"`
	CreatedAt         time.Time  `db:"created_at" json:"created_at"`
	UpdatedAt         time.Time  `db:"updated_at" json:"updated_at"`

	// Joined from users (client)
	ClientName     string `db:"client_name" json:"client_name"`
	ClientCountry  string `db:"client_country" json:"client_country"`
	ClientVerified bool   `db:"client_verified" json:"client_verified"`

	// Computed
	Color          string `db:"-" json:"color"`
	PostedOrder    int    `db:"-" json:"posted_order"`
	AlreadyApplied bool   `db:"-" json:"already_applied"`
}

type JobCreateInput struct {
	Title          string   `json:"title"`
	Summary        string   `json:"summary"`
	Description    string   `json:"description"`
	Category       string   `json:"category"`
	Skills         []string `json:"skills"`
	BudgetType     string   `json:"budget_type"`
	BudgetMin      float64  `json:"budget_min"`
	BudgetMax      float64  `json:"budget_max"`
	DurationLabel  string   `json:"duration_label"`
	ExperienceTier string   `json:"experience_tier"`
	Deliverables   []string `json:"deliverables"`
	Urgent         bool     `json:"urgent"`
	Aspect         *string  `json:"aspect,omitempty"`
}

type JobQuery struct {
	Search     string
	Category   string
	Experience string
	BudgetType string
	Sort       string // "recent" | "budget" | "proposals"
	Page       int
	PageSize   int
}

File: Frontend/types/job.ts
import type { Category, Tier } from "./freelancer"

export type BudgetType = "fixed" | "hourly"

export type JobSort = "recent" | "budget" | "proposals"

export type JobStatus = "open" | "reviewing" | "in_progress" | "completed" | "closed"

export interface Job {
  id: string
  client_id: string
  title: string
  summary: string
  description: string
  category: Category
  skills: string[]
  budget_type: BudgetType
  budget_min: number
  budget_max: number
  duration: string
  experience: Tier | "any"
  deliverables: string[]
  urgent: boolean
  status: JobStatus
  proposals: number
  hired_freelancer_id?: string | null
  aspect?: string | null
  posted_at: string
  created_at: string
  updated_at: string

  // Joined from client user
  client_name: string
  client_country: string
  client_verified: boolean

  // Computed
  color?: string
  posted_order?: number
  already_applied?: boolean
}

export interface JobClient {
  name: string
  country: string
  member_since: string
  verified: boolean
  jobs_posted: number
  hire_rate: number
  rating: number
  reviews: number
  color?: string
}

export interface JobCreateInput {
  title: string
  summary: string
  description: string
  category: Category
  skills: string[]
  budget_type: BudgetType
  budget_min: number
  budget_max: number
  duration_label: string
  experience_tier: Tier | "any"
  deliverables: string[]
  urgent: boolean
  aspect?: string | null
}

export interface JobUpdateInput extends Partial<JobCreateInput> {}

export interface JobQuery {
  search?: string
  category?: Category | ""
  experience?: Tier | ""
  budget_type?: BudgetType | ""
  sort?: JobSort
  page?: number
  page_size?: number
}
